src/viam/trajex/totg/trajectory.cpp

```diff
@@ -1165,38 +1230,64 @@ trajectory trajectory::create(class path p, options opt, integration_points poin
 
+                    // TODO(RSDK-12763): Revisit whether we need epsilon checks before dividing by s_dot_max_vel here. During forward
+                    // integration following the velocity curve, s_dot_max_vel should always be positive. However, if the
+                    // velocity limit itself becomes very small (e.g., tight curvature with low velocity limits), the division
+                    // could become numerically unstable. For now, throw if s_dot_max_vel is near zero as a canary.
+                    if (s_dot_max_vel < traj.options_.epsilon) [[unlikely]] {
+                        // TODO(RSDK-12768): Investigate whether we want a richer exception type for algorithm failures.
+                        // A custom exception hierarchy could distinguish between different failure modes
+                        // (infeasible path, numerical issues, constraint violations) for better error handling.
+                        throw std::runtime_error{
+                            "TOTG algorithm error: cannot evaluate curve exit conditions with near-zero velocity limit"};
+                    }
+
+                    // Check if acceleration limit curve has dropped below velocity limit curve.
+                    // While following the velocity curve, the acceleration curve can drop below us, meaning
+                    // the acceleration constraint becomes the active limit. This is symmetric to the check in
+                    // k_forward_accelerating that detects hitting the velocity curve from below.
+                    if (traj.options_.epsilon.wrap(s_dot_max_acc) < traj.options_.epsilon.wrap(s_dot_max_vel)) {
+                        if (traj.options_.observer) {
+                            traj.options_.observer->on_hit_limit_curve(
+                                traj, {.s = current_point.s, .s_dot = current_point.s_dot}, s_dot_max_acc, s_dot_max_vel);
+                        }
+
+                        // Acceleration curve is now below velocity curve - we've effectively hit the acceleration
+                        // limit curve from above. Need to search for switching point just like Exit B.
+                        auto switching_point = find_switching_point(path_cursor, traj.options_);
+
+                        // Initialize backward integration from switching point
+                        integration_point sp{.time = current_point.time + traj.options_.delta,  // Placeholder, corrected during splice
+                                             .s = switching_point.s,
+                                             .s_dot = switching_point.s_dot,
+                                             .s_ddot = arc_acceleration{0.0}};
+                        backward_points.push_back(std::move(sp));
+
+                        // Reuse k_reached_end event to transition to backward integration
+                        event = integration_event::k_reached_end;
+                        break;
+                    }
```

Comment:
This is important and new in the velocity limit curve following case. Before, we would just charge right through an acceleration limit curve, which we definitely can't do.